src/orange/surf/SoftSurfaceEqual.cc

```diff
@@ -128,11 +141,23 @@ bool SoftSurfaceEqual::operator()(SimpleQuadric const& a,
 bool SoftSurfaceEqual::operator()(GeneralQuadric const& a,
                                   GeneralQuadric const& b) const
 {
-    return &a == &b;
+    return soft_eq_distance_(make_array(a.second()), make_array(b.second()))
+           && soft_eq_distance_(make_array(a.cross()), make_array(b.cross()))
+           && soft_eq_distance_(make_array(a.first()), make_array(b.first()))
+           && soft_eq_(a.zeroth(), b.zeroth());
 }
 
 //---------------------------------------------------------------------------//
 // PRIVATE HELPER FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Compare the square of values for soft equality.
+ */
+bool SoftSurfaceEqual::soft_eq_sq_(real_type a, real_type b) const
+{
+    return soft_eq_(std::sqrt(a), std::sqrt(b));
```

Comment:
Comparing a,b rather than sqrt(a),sqrt(b) is a lot cheaper in terms of computing cycles.  
Also, the comment on L54 says 'Compare the square', not the square-root.

Reply:
I'll update the document. Of course it's cheaper to multiply a number against itself than to calculate the square root.  Unfortunately it's not the same thing to compare the values against epsilon^2 versus comparing the square roots against epsilon. I wanted to put the correct implementation in first, and then once all tests pass I planned to go back to look at microoptimization.